src/pasta/fields/fp.rs

```diff
@@ -646,43 +638,43 @@ impl ff::PrimeField for Fp {
 impl FieldExt for Fp {
     const ROOT_OF_UNITY: Self = ROOT_OF_UNITY;
     const ROOT_OF_UNITY_INV: Self = Fp::from_raw([
-        0x9246674078fa45bb,
-        0xd822ebd60888c5ea,
-        0x56d579133a11731f,
-        0x1c88fa9e942120bb,
+        0xf0b87c7db2ce91f6,
+        0x84a0a1d8859f066f,
+        0xb4ed8e647196dad1,
+        0x2cd5282c53116b5c,
     ]);
     const UNROLL_T_EXPONENT: [u64; 4] = [
-        0x3b3a6633d1897d83,
-        0x0000000000c93d5b,
-        0xf000000000000000,
-        0xe34ab16,
+        0x955a0a417453113c,
+        0x0000000022016b89,
+        0xc000000000000000,
+        0x3f7ed4c6,
     ];
     const T_EXPONENT: [u64; 4] = [
-        0xb61facdcd0a03271,
-        0x0000000001c55093,
+        0x094cf91b992d30ed,
+        0x00000000224698fc,
         0x0000000000000000,
-        0x20000000,
+        0x40000000,
     ];
```

Comment:
I did not check these constants, and I can't find any unit tests for them.

Reply:
See https://github.com/zcash/halo2/pull/84#discussion_r541156767